src/momento/responses/mixins.py

```diff
@@ -0,0 +1,74 @@
+import typing
+
+# https://stackoverflow.com/questions/71889556/mypy-checking-typing-protocol-with-python-3-7-support
+if typing.TYPE_CHECKING:
+    from typing_extensions import Protocol
+else:
+    Protocol = object
+
+from typing import Type, TypeVar
+
+from momento.errors import MomentoErrorCode, SdkException
+
+
+class AbstractClassMixin:
+    """Mixin that simulates an abstract type. Disallows instantiating an object."""
+
+    def __new__(cls, *args, **kwargs):  # type: ignore
+        raise NotImplementedError("This is an abstract class. Cannot instantiate!")
+
+
+class HasValueBytesProtocol(Protocol):
+    @property
+    def value_bytes(self) -> bytes:
+        ...
+
+
+class ValueStringMixin:
+    """Renders `value_bytes` as a utf-8 string.
+
+    Returns:
+        str: the utf-8 encoding of the data
+    """
+
+    @property
+    def value_string(self: HasValueBytesProtocol) -> str:
+        """Convert the bytes `value` to a UTF-8 string
+
+        Returns:
+            str: UTF-8 representation of the `value`
+        """
+        return self.value_bytes.decode("utf-8")
+
+
+class HasErrorProtocol(Protocol):
+    @property
+    def _error(self) -> SdkException:
+        ...
+
+
+TError = TypeVar("TError", bound="ErrorResponseMixin")
+
+
+class ErrorResponseMixin:
+    def __init__(self, _error: SdkException):
+        ...
```

Comment:
Is this necessary? It looks like, unless I'm missing something, removing it doesn't have an effect, in which case I'd prefer to get rid of it.

Reply:
This is necessary so that the `from_sdkexception` method below knows there is an `__init__` method signature that takes an `SdkException`. The actual `__init__` methods are generated by the concrete dataclasses (eg `CacheGetResponse.Error`)